File: src/main/java/com/back/mozu/global/exception/ServiceException.java
package com.back.mozu.global.exception;

import com.back.mozu.global.response.RsData;

public class ServiceException extends RuntimeException {

    private String msg;
    private String resultCode;

    public ServiceException(String resultCode, String msg) {
        super(msg);
        this.msg = msg;
        this.resultCode = resultCode;
    }

    public RsData<Void> getRsData() {
        return new RsData<>(
                msg,
                resultCode
        );
    }
}

File: src/main/java/com/back/mozu/global/exception/ServiceException.kt
package com.back.mozu.global.exception

import com.back.mozu.global.response.RsData

sealed class ServiceException(
    val resultCode: String,
    override val message: String
) : RuntimeException(message) {

    class Common(
        resultCode: String,
        message: String
    ) : ServiceException(resultCode, message)

    fun getRsData(): RsData<Void> {
        return RsData(
            message,
            resultCode
        )
    }
}
